src/topology/constructions.lean

```diff
@@ -496,6 +496,20 @@ lemma continuous_apply [∀i, topological_space (π i)] (i : ι) :
   continuous (λp:Πi, π i, p i) :=
 continuous_infi_dom continuous_induced_dom
 
+/-- Embedding a factor into a product space (by fixing arbitrarily all the other coordinates) is
+continuous. -/
+lemma continuous_update [decidable_eq ι] [∀i, topological_space (π i)] {i : ι} {f : Πi:ι, π i} :
+  continuous (λ x : π i, function.update f i x) :=
+begin
+  refine continuous_pi (λj, _),
+  by_cases h : j = i,
+  { rw [h],
+    simp,
+    exact continuous_id },
```

Comment:
Isn't a non-terminating `simp` considered a bad idea? Does `simp [h, continuous_id]` work?

Reply:
Non-terminating `simp` is indeed considered bad policy. Personally, if it is almost the last instruction in a proof, I don't really mind since it means upon breakage the proof will be easy to fix. The really bad idea is a non-terminating simp in the middle of a long proof.

Here, the three steps are necessary: `simp` and `rw` can not be replaced by each others because of binders, I guess. I have replaced the `simp` by a `simp only`.

Reply:
Sometimes there are other ways. I haven't tested in the above example, and I don't even know what is being proved. But seeing only those three lines I would try to start with `convert continuous_id` before `simp` (or `rw ..., simp` or whatever).

Reply:
I have tried some of those (but certainly not all!). Here, starting with `convert` gives me a goal with the weird equality `==`, which I try to avoid as much as possible.

Some more details on the proof state. At the beginning, we want to show `continuous (λ (a : π i), function.update f i a j)`, and `h` is `h : j = i`. Then `simp [h]` doesn't work, while `rw h` does, leaving me with the goal `continuous (λ (a : π i), function.update f i a i)`. Then `function.update_same` says that `function.update f i a i = f`. Here, `rw [function.update_same]` fails, but `simp only [function.update_same]` works. We are left with the goal ` continuous (λ (a : π i), a)` which is defeq to `continuous_id`, but not syntactically equal, so `simp` or `rw` fails and I need `exact`.

I just realized I could also replace the `simp` and `exact` steps with `simpa [function.update] using continuous_id`, but it feels like cheating. I've done it anyway.

Reply:
I think you spend enough time on those three lines. My remark about `convert` was meant as something to keep in mind when facing such little issues. I think the `simpa` solution is nice.